### DISCUSSION TITLE
Program Info on compilation

### DISCUSSION THREAD

**Discussion by Flofrost** [Alex079/vscode-avr-helper#15] [Ideas]
Greetings,

There's a small feature that's present in AVR Studio that would be very helpful if included in your extension.
When you compile a project in AVR Studio, it tells you how much Flash and SRAM will be taken by the program, both in percent and in bytes.
If you could add the same feature when we build, it would be awesome.
I know the Flash takes is calculated by summing .text + .data + .bootloader, and SRAM by .data + .bss + .noinit (source from "Atmel AVR4027: Tips and Tricks to Optimize your C code for 8-bit AVR Micro controller). These are already available when we compile, but doing the sum by hand is annoying.
I've been using your extension for a while now and it's really great! Thanks you really much for making it!

Sincerely,
         Flofrost

[Tips_and_Tricks.pdf](https://github.com/Alex079/vscode-avr-helper/files/6622479/Tips_and_Tricks.pdf)

**Comment by Alex079** [comment]
Hello!

Thank you for this idea! I will definitely take a look, this sounds like a good improvement.

Regards,
Alex

**Comment by Alex079** [comment]
I checked a bit back then and recently. I think it should be relatively easy to do the sum in bytes. Getting the percentage is a tricky one. The options are
- require using only specific versions of avr-size: Atmel versions support MCU-specific size estimation (not really portable :no_entry_sign:)
- detect which flags are supported by a particular avr-size program and pass the right ones (maybe this is the way to go :ballot_box_with_check:)
- require physical device connection: query MCU memory via avrdude (not really useful for just one quick build, error prone :no_entry_sign:)
- avrdude device config lookup without connection: not available in avrdude directly, although all info is in config file(s) (overly complex :no_entry_sign:)

Maybe there is a better way.
